client/src/main/kotlin/io/spine/examples/pingh/client/VerifyLogin.kt

```diff
@@ -123,48 +126,68 @@ public class VerifyLogin internal constructor(
     }
 
     /**
-     * Checks whether the user has completed the login on GitHub and entered their user code.
+     * Sends requests to verify authentication completion.
      *
-     * Rejections during verification cause the process to transition to the [LoginFailed] stage.
+     * If authentication is not complete, the request is retried
+     * with exponentially increasing intervals.
+     * The process automatically ends when the user code [expires][isUserCodeExpired].
      *
-     * @param onSuccess Called when the login is successfully verified.
-     * @param onFail Called when login verification fails.
+     * @param onSuccess Called when login is verified.
      */
-    public fun confirm(
-        onSuccess: (event: UserLoggedIn) -> Unit = {},
-        onFail: (event: UserIsNotLoggedIntoGitHub) -> Unit = {}
-    ) {
+    public fun waitForAuthCompletion(onSuccess: () -> Unit) {
+        retryStrategy = ExponentialBackoffStrategy.builder()
+            .perform { confirm() }
+            .withMinDelay(interval.asKotlinDuration())
```

Comment:
Let's do the following:

* modify our builder's API to take the original `Duration`,
* make the `Duration.toKotlinDuration()` `private` to `ExponentialBackoffStrategy.kt`,
* rename `Duration.toKotlinDuration` into `Duration.toKotlin`.